### PR TITLE
config: update targets config for v5 compatible targets

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Updates targets for v4 to v5 migration with backwards compatible targets (only targets in a top-level folder).

See https://docs.google.com/document/d/1DOyzvyY7Se19hgVU9f0wSEuMaK7M8vDHz8xHdcPnOjk/edit?resourcekey=0-QU75RQH42NuC-vPNuC1DuQ#

cc @jku 

@haydentherapper for review: I put the new ct log public key under `ctfe_2022.pub` so that it would start with CTFE as well. @vaikas

### CODE REVIEW

**Comment by asraa** [comment]
Nice, e2e tests that used the prod config caught the error I had in path naming, fixed in second commit.

File: config/targets-metadata.yml [hunk 1]
```diff
@@ -52,6 +19,11 @@ ctfe.pub:
     uri: https://ctfe.sigstore.dev
```
Comment (Hayden-IO) [inline config/targets-metadata.yml]: Can we change the URIs to be `https://ctfe.sigstore.dev/test` and `https://ctfe.sigstore.dev/2022`?

Comment (asraa) [reply config/targets-metadata.yml]: All of these are single string key-val pairs. I can dupe the target if need be, are both required?

Comment (Hayden-IO) [reply config/targets-metadata.yml]: Sorry, I meant for `ctfe.pub` -> `/test` and `ctfe_2022.pub` -> `/2022`

Comment (asraa) [reply config/targets-metadata.yml]: done!

Comment (asraa) [reply config/targets-metadata.yml]: note for others: no one relies on the URI fields of custom metadata for discovery.

**Review by Hayden-IO — APPROVED** [review]

**Pull request merged**